addons/mail_plugin/controllers/mail_plugin.py

```diff
@@ -1,263 +1,156 @@
-            if logo_url:
-                try:
-                    response = requests.get(logo_url, timeout=2)
-                    if response.ok:
-                        partner_values.update({'image_1920': base64.b64encode(response.content)})
-                except Exception:
-                    pass
-
-        model_fields_to_iap_mapping = {
-            'street': 'street_name',
-            'city': 'city',
-            'zip': 'postal_code',
-            'website': 'domain',
-        }
-
-        # only update keys for which we dont have values yet
-        partner_values.update({
-            model_field: iap_data.get(iap_key)
-            for model_field, iap_key in model_fields_to_iap_mapping.items() if not partner[model_field]
-        })
-
-        partner.write(partner_values)
-
-        partner.message_post_with_source(
-            'iap_mail.enrich_company',
-            render_values=iap_data,
-            subtype_xmlid='mail.mt_note',
-        )
-
-        return {
-            'enrichment_info': {'type': 'company_updated'},
-            'company': self._get_company_data(partner),
-        }
-
-    @http.route(['/mail_client_extension/partner/get', '/mail_plugin/partner/get']
-        , type="jsonrpc", auth="outlook", cors="*")
-    def res_partner_get(self, email=None, name=None, partner_id=None, **kwargs):
+    @http.route('/mail_plugin/partner/get', type="jsonrpc", auth="outlook", cors="*")
+    def res_partner_get(self, email=None, partner_id=None, **kwargs):
         """
-        returns a partner given it's id or an email and a name.
+        returns a partner given its id or an email.
         In case the partner does not exist, we return partner having an id -1, we also look if an existing company
         matching the contact exists in the database, if none is found a new company is enriched and created automatically
-
-        old route name "/mail_client_extension/partner/get is deprecated as of saas-14.3, it is not needed for newer
-        versions of the mail plugin but necessary for supporting older versions, only the route name is deprecated not
-        the entire method.
         """
-
-        if not (partner_id or (name and email)):
-            return {'error': _('You need to specify at least the partner_id or the name and the email')}
+        if not partner_id and not email:
+            return {'error': _('You need to specify at least the partner_id or the email')}
 
         if partner_id:
             partner = request.env['res.partner'].browse(partner_id).exists()
-            return self._get_contact_data(partner)
+            return self._get_contact_data(partner, email)
 
         normalized_email = tools.email_normalize(email)
         if not normalized_email:
-            return {'error': _('Bad Email.')}
+            return {'error': _('Bad Email.'), 'partner': {}}
 
         notification_emails = request.env['mail.alias.domain'].sudo().search([]).mapped('default_from_email')
```

Comment:
Technically, you should also skip aliases, catchall, ... to avoid creating irrelevant contacts. That is why calling ``_partner_find_from_emails`` could be useful so that all protection done there is done here also (as globally it is the same kind of partner injection).

Or at least use ``self.env['mail.alias.domain'].sudo()._find_aliases``, which globally returns conflicting aliases, catchall, ... and you can return your error, which is what you are looking for here.